src/editors/select.js

```diff
@@ -39,7 +39,7 @@ export var SelectEditor = AbstractEditor.extend({
     return Math.min(12,Math.max(longest_text/7,2));
   },
   typecast: function(value) {
-    if (this.schema.type === "boolean") return !!value;
+    if (this.schema.type === "boolean") return value == 'undefined' || value === undefined ? undefined : !!value;  
     else if (this.schema.type === "number") return 1*value || 0;
```

Comment:
and `Number(value);` here?

Reply:
That's not enough, as the number typecasting is potentially buggy too. It would also have to check for `Infinity` to be safe. So it should be something like:
````javascript
if (this.schema.type === "number") return value === Infinity ? 0 : value*1 || 0;  
````
Or:
````javascript
if (this.schema.type === "number") return value === Infinity ? 0 : Number(value) || 0; 
````

As:
Number(1/0) = Infinity
Number(1/0) || 0 = Infinity

Number("abc") = NaN
Number("abc") || 0 = 0

Number("12") = 12
Number("12") || 0 = 12